app/src/main/java/com/evanfuhr/pokemondatabase/activities/PokemonDisplayActivity.java

```diff
@@ -116,6 +130,116 @@ private void setPokemonBackgroundColor(Pokemon pokemon) {
+        // automatically handle clicks on the Home/Up button, so long
+        // as you specify a parent activity in AndroidManifest.xml.
+        int id = item.getItemId();
+
+        switch(id) {
+            case R.id.action_set_game:
+                onClickMenuSetGame();
+                break;
+            case R.id.action_search_list:
+                break;
+            default:
+                return super.onOptionsItemSelected(item);
+        }
+
+        return true;
+    }
+
+    public void onClickMenuSetGame() {
+
+        // Get list of versions
+        VersionDAO versionDAO = new VersionDAO(this);
+        List<Version> versionList = versionDAO.getAllVersions();
+
+        // Get view
+        LayoutInflater layoutInflater = LayoutInflater.from(PokemonDisplayActivity.this);
+        View setGameVersionView = layoutInflater.inflate(R.layout.dialog_set_game_version, null);
+
+        // Setup spinner
+        Spinner versionSpinner = setGameVersionView.findViewById(R.id.spinner_game_version);
+        final VersionSpinnerAdapter versionSpinnerAdapter = new VersionSpinnerAdapter(this, R.layout.dialog_set_game_version, versionList);
+        versionSpinner.setAdapter(versionSpinnerAdapter);
+        // TODO: Set position
+        restorePreferences();
+        versionSpinner.setSelection(versionSpinnerAdapter.getPositionByVersion(mVersionId));
+
+        versionSpinner.setOnItemSelectedListener(new AdapterView.OnItemSelectedListener() {
+            @Override
+            public void onItemSelected(AdapterView<?> parent, View view, int position, long id) {
+                Version version = versionSpinnerAdapter.getItem(position);
+                mVersionId = version.getId();
+
+            }
+
+            @Override
+            public void onNothingSelected(AdapterView<?> parent) {
+
+            }
+        });
+
+        // Setup a dialog window
+        AlertDialog.Builder alertDialogBuilder = new AlertDialog.Builder(PokemonDisplayActivity.this);
+        alertDialogBuilder.setTitle(R.string.set_game_version);
+        alertDialogBuilder.setView(setGameVersionView);
+        alertDialogBuilder.setCancelable(true)
+                .setPositiveButton(R.string.save, new DialogInterface.OnClickListener() {
+                    public void onClick(DialogInterface dialog, int id) {
+                        // Save game version
+                        dialog.dismiss();
+
+                        // We need an Editor object to make preference changes.
+                        // All objects are from android.context.Context
+                        SharedPreferences settings = getSharedPreferences(String.valueOf(R.string.gameVersionID), MODE_PRIVATE);
+                        SharedPreferences.Editor editor = settings.edit();
+                        editor.clear();
+                        editor.putInt(String.valueOf(R.string.gameVersionID), mVersionId);
+                        editor.commit();
```

Comment:
Lines 201 to 209 look like they can be moved into a separate inner class or anonymous class since it looks like its happening in multiple places?

Reply:
I think i can put this in a utility function